layers/b2qt/recipes-qt/automotive/neptune-ui_git.bbappend

```diff
@@ -10,3 +10,40 @@ RDEPENDS_${PN}_append = "\
       qtwebengine        \
       qtquickcontrols2   \
       "
+
+HAS_CONTAINMENT = "${@bb.utils.contains('DISTRO_FEATURES', 'process-containment', 'TRUE', 'FALSE', d)}"
+
+#
+# Add software-container AM config to appman cmdline if we have containment
+# support
+#
+do_install_prepend() {
+    echo "Containment: ${HAS_CONTAINMENT}"
+    SC_CONFIG_LINE="-c /opt/am/sc-config.yaml"
+    SERVICEFILE="${WORKDIR}/neptune.service"
+
+
+    # We only add it if we have containment and it isn't already there
+    if [ "${HAS_CONTAINMENT}" = "TRUE" ]; then
+        if  ! grep -q "sc-config.yaml" $SERVICEFILE; then
+            echo "Adding SC config line"
+            sed -i -e "s|^ExecStart=\(.*\)$|ExecStart=\1 $SC_CONFIG_LINE|" $SERVICEFILE
+        fi
+    else
+	# Remove any sc configs from the config file
+        echo "Removing SC line if present"
+        sed -i -e "s|$SC_CONFIG_LINE||" $SERVICEFILE
```

Comment:
Why do you need to remove it ? If the DISTRO doesn't have the feature, it shouldn't be there in the first place ? And if it is, the user might know what he is doing ?

Reply:
Sure, fair point.